trinity/tx_pool/validators.py

```diff
@@ -0,0 +1,83 @@
+from abc import (
+    ABC,
+    abstractmethod
+)
+from typing import (
+    Type
+)
+
+from evm.chains.base import (
+    BaseChain
+)
+from evm.exceptions import (
+    ValidationError
+)
+from evm.rlp.transactions import (
+    BaseTransaction,
+    BaseTransactionFields
+)
+from evm.vm.base import (
+    BaseVM
+)
+
+
+class BaseTransactionValidator(ABC):
+    """
+    The :class:`~trinity.tx_pool.validators.BaseTransactionValidator` class is the base class for
+    transaction validation done by the :class:`~trinity.tx_pool.pool.TxPool`.
+
+        .. note::
+        This is an abstract class that must be implemented in subclasses
+    """
+
+    @abstractmethod
+    def is_valid(self, transaction: BaseTransactionFields) -> bool:
```

Comment:
classes with a single method are generally an anti-pattern for me.  Instead of using a class for this, can we instead just change to just passing in a callable?  

We can still implement more complex validation using classes if desired and just implement `__call__` for the check.

Reply:
I generally agree and I had it a plain callable in the beginning but then changed it to a class because I wanted the tx pool to provide a default implementation that takes constructor arguments. We can also solve that using currying if you prefer.

Reply:
Ah, or the default one continues to be a class and implements `__call__` as you suggests and we just drop the abstract class in favor for the more lightweight `Callable[[TX], bool]` interface. I guess that would be my preference.

Reply:
I'm good with either a class with `__call__` **or** a curried callable.

Reply:
Yep, I pushed the call based solution